src/utils.py

```diff
@@ -69,3 +83,64 @@ def print_os_environment():
     """
     for key, val in os.environ.items():
         print(f"{key}: {val}")
+
+
+def get_semver_user_input(body):
+    match = re.match(
+        rf".*\B/(?P<{SEMVER}>{LABEL_KEY_MAJOR}|{LABEL_KEY_MINOR}|{LABEL_KEY_PATCH})\b",
+        body,
+        re.IGNORECASE,
+    )
+    if match:
+        return match.group(SEMVER).lower()
+
+
+def semver_labels_in_pull_labels(pull):
+    return [
+        label in SEMVER_LABELS_LIST
+        for label in get_labels(pull=pull)
+        if label in SEMVER_LABELS_LIST
+    ]
+
+
+def add_remove_labels(pull, labels_to_add, labels_to_remove):
+    label_removed = False
+    new_labels_to_add = labels_to_add.copy()
+    labels_from_pull = get_labels(pull=pull)
+
+    for label in labels_from_pull:
+        if label in labels_to_remove:
+            remove_label(pull=pull, label=label)
+            label_removed = True
+        elif label in new_labels_to_add:
+            print(f"Label {label} already exists in {pull.title}")
+            new_labels_to_add.remove(label)
+
+    for label in new_labels_to_add:
+        add_label(pull=pull, label=label)
+
+    return new_labels_to_add or label_removed
+
+
+def get_semver_label_data(pull, body):
+    labels_to_add = []
+    labels_to_remove = []
+    semver_key = get_semver_user_input(body=body)
+    existing_semver_labels = semver_labels_in_pull_labels(pull=pull)
+
+    if semver_key or existing_semver_labels:
+        if (
+            semver_key
+            and SEMVER_USER_INPUT_LABELS_MAP[semver_key] not in existing_semver_labels
+        ):
+            labels_to_add = [SEMVER_USER_INPUT_LABELS_MAP[semver_key]]
+            labels_to_remove = list(
+                SEMVER_LABELS_LIST - {SEMVER_USER_INPUT_LABELS_MAP[semver_key]}
+            )
+        status_state = STATE_SUCCESS
+        status_description = STATUS_DESCRIPTION_SEMVER_LABELS_EXIST
+    else:
+        status_state = STATE_PENDING
+        status_description = STATUS_DESCRIPTION_SEMVER_LABELS_MISSING
```

Comment:
```
 status_state = STATE_SUCCESS
        status_description = STATUS_DESCRIPTION_SEMVER_LABELS_EXIST
    else:
        status_state = STATE_PENDING
        status_description = STATUS_DESCRIPTION_SEMVER_LABELS_MISSING
```
Not belong to this function, should be set when adding/remove labels.

Reply:
https://github.com/myakove/pr-workflows/pull/33/files#r708220081